plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-button/payment-gateway-button.tsx

```
@@ -1,88 +1,69 @@
-
-	const toggleEnabled = async () => {
-		setIsLoading( true );
-
-		if ( ! window.woocommerce_admin.nonces?.gateway_toggle ) {
-			// eslint-disable-next-line no-console
-			console.warn( 'Unexpected error: Nonce not found' );
-			// Redirect to payment setting page if nonce is not found. Users should still be able to toggle the payment method from that page.
-			window.location.href = settings_url;
-			return;
-		}
-
-		try {
-			const response = await fetch( window.woocommerce_admin.ajax_url, {
-				method: 'POST',
-				headers: {
-					'Content-Type': 'application/x-www-form-urlencoded',
-				},
-				body: new URLSearchParams( {
-					action: 'woocommerce_toggle_gateway_enabled',
-					security: window.woocommerce_admin.nonces?.gateway_toggle,
-					gateway_id: id,
-				} ),
-			} );
+	const { enablePaymentGateway } = useDispatch(
+		PAYMENT_GATEWAYS_SUGGESTIONS_STORE_NAME
+	);
 
-			const result = await response.json();
-
-			if ( result.success ) {
-				if ( result.data === true ) {
-					setIsEnabled( true );
-				} else if ( result.data === false ) {
-					setIsEnabled( false );
-				} else if ( result.data === 'needs_setup' ) {
-					setNeedsSetup( true );
-					window.location.href = settings_url;
-				}
-			} else {
-				window.location.href = settings_url;
-			}
-		} catch ( error ) {
-			// eslint-disable-next-line no-console
-			console.error( 'Error toggling gateway:', error );
-		} finally {
-			setIsLoading( false );
-		}
-	};
+	const { isUpdating, shouldRedirect } = useSelect( ( select ) => {
+		return {
+			isUpdating: select(
+				PAYMENT_GATEWAYS_SUGGESTIONS_STORE_NAME
+			).isUpdating( id ),
+			shouldRedirect: select(
+				PAYMENT_GATEWAYS_SUGGESTIONS_STORE_NAME
+			).shouldRedirect( id ),
+		};
+	} );
 
 	const onClick = ( e: React.MouseEvent ) => {
 		if ( ! enabled ) {
 			e.preventDefault();
-			toggleEnabled();
+			const gateway_toggle_nonce =
+				window.woocommerce_admin.nonces?.gateway_toggle || '';
```

If `window.woocommerce_admin.nonces?.gateway_toggle` is empty, should we throw an error instead of using an empty string since it will cause the API call to fail?

---

yeah, agree! Could you give an advice how to throw it properly? please - it's one thing left I resolved all other and I feel done for today 🙈

---

I think we can show an error notice that the nonce is missing and suggest refreshing the page (or any other action that the user can try). 🙂 

- https://developer.wordpress.org/block-editor/reference-guides/data/data-core-notices/#createerrornotice

---

fixed in https://github.com/woocommerce/woocommerce/pull/52824/commits/8d6c915eb8cf7aeabf504a0c909b6f527fb530d8